### PR TITLE
[ISSUE-216] Support sized chunk fetch request

### LINKED ISSUE #216: [FEATURE] Support sized chunk fetch request
### Describe alternatives you've considered

/cc @who-need-to-know

/assign @who-can-help-you

### PR DESCRIPTION
**Pull request by waitinfuture**
# [BUG]/[FEATURE] title

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

close #216

### CODE REVIEW

File: common/src/main/java/com/aliyun/emr/rss/common/network/server/FileManagedBuffers.java [hunk 1]
```diff
@@ -66,23 +62,27 @@ public boolean hasAlreadyRead(int chunkIndex) {
...
+    final long chunkOffset = offsets[chunkIndex];
+    final long chunkLength = offsets[chunkIndex + 1] - chunkOffset;
+    assert offset < chunkLength;
+    long length = Math.min(chunkLength - offset, len);
+    if (len + offset >= chunkLength) {
+      synchronized (chunkTracker) {
+        chunkTracker.set(chunkIndex);
+      }
+      if (chunkIndex == numChunks - 1) {
+        fullyRead = true;
```
Comment (AngersZhuuuu) [inline common/src/main/java/com/aliyun/emr/rss/common/network/server/FileManagedBuffers.java:79]: In current logic seems we have not considered the case that some chunks read partial data?

Comment (waitinfuture) [reply common/src/main/java/com/aliyun/emr/rss/common/network/server/FileManagedBuffers.java:79]: It's under the condition len + offset >= chunkLength, which means the current chunk is fully read.

**Comment by FMX** [comment]
LGTM.

**Comment by FMX** [comment]
LGTM.

**Review by FMX — APPROVED** [review]

**Pull request merged**